tests/php/Unit/Analyzer/TupleSymbol/IfSymbolTest.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhelTest\Unit\Analyzer\TupleSymbol;
+
+use Generator;
+use Phel\Analyzer;
+use Phel\Analyzer\TupleSymbol\IfSymbol;
+use Phel\Ast\IfNode;
+use Phel\Exceptions\PhelCodeException;
+use Phel\GlobalEnvironment;
+use Phel\Lang\Symbol;
+use Phel\Lang\Tuple;
+use Phel\NodeEnvironment;
+use PHPUnit\Framework\TestCase;
+
+final class IfSymbolTest extends TestCase
+{
+    /**
+     * @dataProvider providerRequiresAtLeastTwoOrThreeArgs
+     */
+    public function testRequiresAtLeastTwoOrThreeArgs(Tuple $tuple): void
+    {
+        $this->expectException(PhelCodeException::class);
+        $this->expectExceptionMessage("'if requires two or three arguments");
+
+        $this->analyze($tuple);
+    }
+
+    public function providerRequiresAtLeastTwoOrThreeArgs(): Generator
+    {
+        yield 'No arguments provided: (if)' => [
+            'tuple' => Tuple::create(
+                Symbol::create(Symbol::NAME_IF)
+            ),
+        ];
+
+        yield 'Only one argument provided: (if one)' => [
+            'tuple' => Tuple::create(
+                Symbol::create(Symbol::NAME_IF),
+                Symbol::create('one'),
```

Comment:
@jenshaase I got a question for you here: is it really necessary to decorate these literal values as Symbols? I mean, for example, the test will still pass(green) with these:
```php
'tuple' => Tuple::create(
    Symbol::NAME_IF,
    'one',
),
// OR
'tuple' => Tuple::create(
    Symbol::NAME_IF,
    'one',
    'two',
    'three',
    'four',
),
```

Reply:
Yes you can do this. It doesn't matter for the test. But it is different Phel code and you may want to update it in the string above.

```
(if one two three four)
```
vs
```
(if "one" "two" "three" "four")
```